docs/GettingStarted.md

```diff
@@ -0,0 +1,139 @@
+# Getting Started
+
+CYNQ is as straight-forward as PYNQ. The philosophy behind CYNQ is to be as simple as PYNQ.
+
+1) The first step to integrate CYNQ is to include the header:
+
+```c++
+#include <cynq/cynq.hpp>
+```
+
+2) Create a IHardware object that will generate the proper drivers for handling the accelerator, data mover and memory. It uses the `IHardware::Create(impl, bitstream, xclbin)` factory. For instance:
+
+```c++
+auto impl = cynq::HardwareArchitecture::UltraScale;
+auto platform = 
+    cynq::IHardware::Create(impl, bitstream, xclbin);
+```
+
+where:
+
+* `impl` is the HardwareArchitecture. At the moment, only `UltraScale` is supported for UltraScale MPSoC.
+* `bitstream` is the path to the .bit file (bitstream)
+* `xclbin` is the path to the .xclbin file. You can use the one in `third-party/resources/default.xclbin`.
+
+3) Create the DMA instances to move the data. This is intended for designs that uses AXI4-Stream.
+
+```c++
+constexpr int kDmaAddress = 0xa0010000;
+auto dma = platform->GetDataMover(kDmaAddress);
+```
+
+where `kDmaAddress` is the address of the DMA instance we want to control. This is given by the design.
+
+4) Create the IP core instances to interact with them.
+
+```c++
+constexpr int kAccelAddress = 0xa0000000;
+auto accel = platform->GetAccelerator(kAccelAddress);
+```
+
+where `kAccelAddress` is the AXI4-Lite control port of the IP core. As a requirement:
+
+> The IP Cores must have an AXI4-Lite port as a control port
+
+5) Get buffers to exchange data. These buffers are usually dual memory: they are mapped into host and device regions (physically contiguous).
+
+```c++
+std::size_t input_size = 64; // bytes
+std::size_t output_size = 64; // bytes
+auto type = MemoryType::Dual; // dual memory
+
+auto in_mem = dma->GetBuffer(input_size, type);
+auto out_mem = dma->GetBuffer(output_size, type);
+```
+
+where the `GetBuffer()` method includes: `size` in bytes and `type` of the memory type:
+
+* `cynq::MemoryType::Dual`: allocates two memory regions, one accessible from host and another from device.
+* `cynq::MemoryType::Cacheable`: allocates a memory region which is cacheable.
+* `cynq::MemoryType::Host`: allocates a host-only memory.
+* `cynq::MemoryType::Device`: allocates a device-only memory.
+
+6) To access the data from the memory buffers, you can use the `HostAddress` method.
+
+```c++
+using DataType = uint64_t;
+DataType* A = in_mem->HostAddress<DataType>().get();
+
+A[5] = 1;
+```
+
+The `HostAddress<T>()` maps the memory into a pointer that is accessible to the host. `T` can be any type that can be reinterpretedly casted.
```

Comment:
any type that supports reinterpret_pointer_cast and provide link to the cpp reference doc this is meant for newbies.
https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast